src/hooks/useApi.js

```diff
@@ -0,0 +1,24 @@
+import { useState } from 'react'
```

Comment:
General thought: 
I got confused with the changes in DetailsPage and the states used there which get set in this custom hook. It makes sense though, but maybe a short comment about the usage of the custom hook could be nice. Maybe in a js readable doc style (/** */), since it is a generic feature which might gets reused a couple of times. The comment gets shown while hovering the custom Hook and saves time you would spend jumping back in the hooks file itself

Reply:
Yes good point i will add some comments about what is happening